feature/experimental/bgp/ate_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -0,0 +1,379 @@
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+	iDut1 := topo.AddInterface(ateAttrs.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateAttrs.IPv4CIDR()).WithDefaultGateway(dutAttrs.IPv4)
+	bgpDut1 := iDut1.BGP()
+	bgpPeer := bgpDut1.AddPeer().WithPeerAddress(dutAttrs.IPv4).WithLocalASN(ateAS).WithTypeExternal()
+
+	t.Log("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	// Verify Port Status
+	t.Log("Verifying port status")
+	verifyPortsUp(t, dut.Device)
+
+	// Verify BGP status
+	t.Log("Check BGP parameters")
+	verifyBgpTelemetry(t, dut)
+
+	// Verify BGP capabilities
+	t.Log("Check BGP Capabilities")
+	verifyBGPCapabilities(t, dut)
+
+	// Send Cease Notification from ATE to DUT
+	t.Log("Send Cease Notification from ATE to DUT")
+	ate.Actions().NewBGPPeerNotification().WithCode(6).WithSubCode(6).WithPeers(bgpPeer).Send(t)
+
+	// Verify BGP session state : ACTIVE
+	t.Log("Verify BGP session state : ACTIVE")
+	nbrPath.SessionState().Await(t, time.Second*60, telemetry.Bgp_Neighbor_SessionState_ACTIVE)
+
+	// Verify if Cease notification is received on DUT.
+	t.Log("Verify Error code received on DUT: BgpTypes_BGP_ERROR_CODE_CEASE")
+	code := nbrPath.Messages().Received().LastNotificationErrorCode().Get(t)
+	if code != telemetry.BgpTypes_BGP_ERROR_CODE_CEASE {
+		t.Errorf("On disconnect: expected error code %v, got %v", telemetry.BgpTypes_BGP_ERROR_CODE_CEASE, code)
+	}
+
+	// Clear config on DUT and ATE
+	topo.StopProtocols(t)
+	dutConfPath.Replace(t, nil)
+}
+
+// TestParameters is to verify normal session establishment and termination
+// in both eBGP and iBGP scenarios using session parameters like explicit
+// router id , timers and MD5 authentication.
+func TestParameters(t *testing.T) {
+	ateIP := ateAttrs.IPv4
+	dutIP := dutAttrs.IPv4
+	dut := ondatra.DUT(t, "dut")
+
+	// Configure Network instance type and router id on DUT
+	t.Log("Configure Network Instance")
+	dutConfNIPath := dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance)
+	dutConfNIPath.Type().Replace(t, telemetry.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+	dutConfNIPath.RouterId().Replace(t, dutAttrs.IPv4)
+
+	dutConfPath := dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	statePath := dut.Telemetry().NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateIP)
+
+	cases := []struct {
+		name      string
+		dutConf   *telemetry.NetworkInstance_Protocol_Bgp
+		ateConf   *ondatra.ATETopology
+		wantState *telemetry.NetworkInstance_Protocol_Bgp
+	}{
+		{
+			name: "Test the eBGP session establishment: Global AS ",
+			dutConf: bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS,
+				routerID: dutAttrs.IPv4, holdTime: 100, connRetTime: 100, authPwd: "AUTHPASSWORD"}),
```

Comment:
If the other parameters like routerID, holdTime, connRetTime and authPwd are all the same now, just inline them into bgpCreateNbr and omit them from bgpTestParams?